crates/re_viewer/src/app.rs

```diff
@@ -886,25 +875,39 @@ impl App {
             return false;
         }
 
+        let sources = self.rx.sources();
+
+        if sources.is_empty() {
+            return true;
+        }
+
         // Here, we use the type of Receiver as a proxy for which kind of workflow the viewer is
         // being used in.
-        match self.rx.source() {
-            // These source are typically "finite". We want the loading screen so long as data is
-            // coming in.
-            SmartChannelSource::Files { .. } | SmartChannelSource::RrdHttpStream { .. } => {
-                !self.rx.is_connected()
+        for source in sources {
+            match &*source {
+                // These source are typically "finite". We want the loading screen so long as data is
+                // coming in.
+                SmartChannelSource::File(_) | SmartChannelSource::RrdHttpStream { .. } => {
+                    return true;
```

Comment:
This breaks the current behaviour where the web demo would show the loading screen instead of the welcome screen until the recording is shown. Now the welcome screen flashes before the recording is shown, with is not super nice.

Reply:
Good catch!